src/components/menus/ExtensionMenu.svelte

```diff
@@ -0,0 +1,68 @@
+<svelte:options accessors={true} immutable={true} />
+
+<script lang="ts">
+  import SettingsIcon from '@nasa-jpl/stellar/icons/settings.svg?component';
+  import { createEventDispatcher } from 'svelte';
+  import type { User } from '../../types/app';
+  import type { Extension } from '../../types/extension';
+  import { permissionHandler } from '../../utilities/permissionHandler';
+  import PlanNavButton from '../plan/PlanNavButton.svelte';
+  import MenuItem from './MenuItem.svelte';
+
+  export let extensions: Extension[];
+  export let user: User | null;
+
+  const dispatch = createEventDispatcher();
+
+  function callExtension(extension: Extension) {
+    dispatch('callExtension', extension);
+  }
+
+  function hasExtensionPermission(extension: Extension): boolean {
+    for (const extensionRole of extension.extension_roles) {
+      if (user?.activeRole === extensionRole.role) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+</script>
+
+{#if extensions.length > 0}
+  <div class="view-menu st-typeography-medium">
+    <PlanNavButton title="Extensions" menuTitle="">
```

Comment:
Any reason not to have a menu title here?

Reply:
Nope, I didn't realize what it was for on my first pass but now I see it shows in the dropdown. I'll add one!